precice/SolverInterface.hpp and some more headers not found

**Describe your setup**

Operating system (e.g. Linux distribution and version): _Ubuntu 20.04_
preCICE Version: My working state at [16f0361](https://github.com/precice/precice/pull/1523/commits/16f03617c142e4c0e514d95245c37d44c6641cf2), but should also happen for [8941459](https://github.com/precice/precice/commit/89414592d7ada5056bcac068d1c99641fdd33e4f).

**Describe the problem**

I usually install preCICE by running `sudo make install` from the build directory. If after installation I look at `pkg-config --cflags libprecice` I get `INCLUDE_PATH` (in my case `/usr/local/include`). Looking at `/usr/local/include/precice/` I get the following:

```
$ ls /usr/local/include/precice/
export.h  SolverInterfaceC.h  SolverInterfaceFortran.hpp
```

If I try (for example) to build the python bindings at this point, I get several errors, because `precice/SolverInterface.hpp` is missing:

```
...
  /tmp/pip-req-build-hk808dta/cyprecice/cyprecice.cpp:810:10: fatal error: precice/SolverInterface.hpp: No such file or directory
    810 | #include "precice/SolverInterface.hpp"
...
```

Manually copying several files helps me to fix this situation:

```
precice/build$ sudo cp ../src/precice/SolverInterface.hpp /usr/local/include/precice/.
precice/build$ sudo cp src/precice/Version.h /usr/local/include/precice/.
precice/build$ sudo cp ../src/precice/Tooling.hpp /usr/local/include/precice/.
```

**Step To Reproduce**
1. Before installation, make sure that your `INCLUDE_PATH/precice` is empty and does not contain files from previous installations.
2. Install preCICE
3. Check `INCLUDE_PATH/precice`. Does `SolverInterface.hpp` exist?

**Expected behaviour**
Installing preCICE via `sudo make install` should take care of copying `SolverInterface.hpp`, `Version.hpp` and `Tooling.hpp` to the include directory.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -371,7 +371,7 @@ GENERATE_EXPORT_HEADER(preciceCore
   )
 
 # Add the export header to the public headers of the preCICE lib
-set_property(TARGET precice PROPERTY PUBLIC_HEADER
+set_property(TARGET precice APPEND PROPERTY PUBLIC_HEADER
     ${CMAKE_BINARY_DIR}/src/precice/export.h)
 
 #

